google/cloud/sql/connector/instance_connection_manager.py

```diff
@@ -371,12 +371,40 @@ async def _perform_refresh(self) -> asyncio.Task:
 
         logger.debug("Entered _perform_refresh")
 
-        self._current = self._loop.create_task(self._get_instance_data())
-        # Ephemeral certificate expires in 1 hour, so we schedule a refresh to happen in 55 minutes.
-
-        self._next = self._loop.create_task(self._schedule_refresh())
-
-        return self._current
+        refresh_task = self._loop.create_task(self._get_instance_data())
+
+        async def _refresh_callback(task: asyncio.Task) -> None:
+            try:
+                task.result()
+            except Exception as e:
+                logger.warn(
```

Comment:
Do we really want to log this, or would it be better to include the context in the Exception?

Reply:
We're logging it in the Java Connector: https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/main/core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java#L381-L382

Calling `task.result` on a completed task will either return the `InstanceData` object or throw an error. This `try/catch` block is the equivalent of defining the `onSuccess` and `onFailure` in the Java `FutureCallback`